IBPSA/Fluid/Sources/MassFlowSourceFromOutside_h.mo

```diff
@@ -0,0 +1,202 @@
+    "Prescribed mass flow rate"
+    annotation (Placement(transformation(extent={{-120,60},{-80,100}}), iconTransformation(extent={{-120,60},{-80,100}})));
+  Modelica.Blocks.Interfaces.RealInput C_in[Medium.nC](
+    final quantity=Medium.extraPropertiesNames) if use_C_in
+    "Prescribed boundary trace substances"
+    annotation (Placement(transformation(extent={{-120,-100},{-80,-60}})));
+  IBPSA.BoundaryConditions.WeatherData.Bus weaBus "Bus with weather data"
+    annotation (Placement(transformation(extent={{-110,-10},{-90,10}}),
+        iconTransformation(extent={{-120,-18},{-80,22}})));
+
+protected
+  Modelica.SIunits.Conversions.NonSIunits.Temperature_degC TDryBul_degC
+    "Dry bulb temperature in degree Celsius";
+  Modelica.SIunits.Pressure p_w(displayUnit="Pa")  "Water vapor pressure";
+  Modelica.SIunits.MassFraction XiDryBul(nominal=0.01)
+    "Water vapor mass fraction at dry bulb state";
+  Modelica.Blocks.Interfaces.RealOutput TDryBul(final unit="K", displayUnit="degC")
+    "Needed to calculate specific enthalpy";
+  Modelica.Blocks.Interfaces.RealOutput relHum(final unit="1")
+    "Needed to calculate specific enthalpy";
+  Modelica.Blocks.Interfaces.RealOutput pAtm(final unit="Pa")
+    "Needed to calculate specific enthalpy";
+  Modelica.Blocks.Interfaces.RealOutput h_in_calculation(final unit="J/kg")
+    "Needed to calculate specific enthalpy";
+  Modelica.Blocks.Interfaces.RealOutput h_out_internal(final unit="J/kg")
+    "Needed to connect to conditional connector";
+
+  final parameter Boolean singleSubstance = (Medium.nX == 1)
+     "True if single substance medium";
+  IBPSA.Utilities.Psychrometrics.X_pTphi x_pTphi if (not singleSubstance)
+      "Block to compute water vapor concentration";
+  Modelica.Blocks.Interfaces.RealOutput m_flow_in_internal(final unit="kg/s")
+    "Needed to connect to conditional connector";
+  Modelica.Blocks.Interfaces.RealInput h_in_internal(final unit="J/kg")
+    "Needed to connect to conditional connector";
+  Modelica.Blocks.Interfaces.RealOutput X_in_internal[Medium.nX](
+    each final unit = "kg/kg",
+    final quantity=Medium.substanceNames)
+    "Needed to connect to conditional connector";
+  Modelica.Blocks.Interfaces.RealOutput C_in_internal[Medium.nC](
+    final quantity=Medium.extraPropertiesNames)
+    "Needed to connect to conditional connector";
+
+equation
+  Modelica.Fluid.Utilities.checkBoundary(
+    Medium.mediumName,
+    Medium.substanceNames,
+    Medium.singleState,
+    true,
+    medium.X,
+    "MassFlowSourceFromOutside_h");
+
+  // Connections and calculation to find specific enthalpy
+  connect(weaBus.pAtm, pAtm);
+  connect(weaBus.TDryBul, TDryBul);
+  connect(weaBus.relHum, relHum);
+  TDryBul_degC = TDryBul - 273.15;
+  p_w = relHum * IBPSA.Utilities.Psychrometrics.Functions.saturationPressure(TDryBul);
+  XiDryBul = 0.6219647130774989*p_w/(pAtm-p_w);
+  h_in_calculation = 1006*TDryBul_degC + XiDryBul*(2501014.5+1860*TDryBul_degC);
```

Comment:
Use the enthalpy calculation from the `Medium` instance -- otherwise, if someone uses a different model for air, or uses a model that has a different value for the 0 enthalpy, the results will be wrong.